omp/base/index_set_kernels.cpp

```diff
@@ -181,30 +181,26 @@ GKO_INSTANTIATE_FOR_EACH_INDEX_TYPE(
 
 template <typename IndexType>
 void local_to_global(std::shared_ptr<const DefaultExecutor> exec,
-                     const IndexType index_space_size,
-                     const Array<IndexType>* subset_begin,
-                     const Array<IndexType>* subset_end,
-                     const Array<IndexType>* superset_indices,
-                     const Array<IndexType>* local_indices,
-                     Array<IndexType>* global_indices, const bool is_sorted)
+                     const IndexType num_subsets, const IndexType* subset_begin,
+                     const IndexType* superset_indices,
+                     const IndexType num_indices,
+                     const IndexType* local_indices, IndexType* global_indices,
+                     const bool is_sorted)
 {
 #pragma omp parallel for
-    for (size_type i = 0; i < local_indices->get_num_elems(); ++i) {
-        auto index = local_indices->get_const_data()[i];
-        GKO_ASSERT(
-            index <=
-            (superset_indices
-                 ->get_const_data()[superset_indices->get_num_elems() - 1]));
-        auto bucket = std::distance(
-            superset_indices->get_const_data(),
-            std::upper_bound(superset_indices->get_const_data(),
-                             superset_indices->get_const_data() +
-                                 superset_indices->get_num_elems(),
-                             index));
+    for (size_type i = 0; i < num_indices; ++i) {
+        auto index = local_indices[i];
+        if (index >= superset_indices[num_subsets]) {
+            global_indices[i] = invalid_index<IndexType>();
+            continue;
+        }
+        const auto bucket = std::distance(
+            superset_indices,
+            std::upper_bound(superset_indices,
+                             superset_indices + num_subsets + 1, index));
         auto shifted_bucket = bucket == 0 ? 0 : (bucket - 1);
```

Comment:
No need for the shift if you already use shifted inputs
```suggestion
        const auto bucket = std::distance(
            superset_indices + 1,
            std::upper_bound(superset_indices + 1,
                             superset_indices + num_subsets + 1, index));
```